src/screens/Profile/Login/ConfirmCode.tsx

```diff
@@ -64,6 +69,20 @@ export default function ConfirmCode({navigation, route}: ConfirmCodeProps) {
     }
   };
 
+  // User might be automatically logged in with Firebase auth, but only on Android
+  // Check user from state and see if it is updated while we wait
+  useEffect(() => {
+    if (user?.phoneNumber) {
```

Comment:
I would prefer to let `authenticationType` instead of `user` to decide this. User is a FirebaseAuth internal type that I think we should use sparingly in other parts of the code. I have been thinking of removing if from the returned auth state, just haven't got time yet.

Reply:
Fixed.